### PR TITLE
IPZ:API to read IPZ keyword from hardware

### PR DESCRIPTION
**Pull request by PriyangaRamasamy**
This commit implements API to read keyword value from hardware in IpzVpdParser class.

Test:
Test works as expected.

### CODE REVIEW

File: src/ipz_parser.cpp [hunk 1]
```diff
@@ -440,4 +440,80 @@ types::VPDMapVariant IpzVpdParser::parse()
...
+    if (const types::IpzType* l_ipzData =
+            std::get_if<types::IpzType>(&i_paramsToReadData))
+    {
+        l_record = std::get<0>(*l_ipzData);
+        l_keyword = std::get<1>(*l_ipzData);
+    }
+    else
+    {
+        logging::logMessage("Given VPD is not of type IPZ. Abort read.");
+        throw types::DeviceError::ReadFailure();
```
Comment (SunnySrivastava1984) [inline src/ipz_parser.cpp:481]: Why it is a device error if data set by the user is wrong? I would suggest a generic runtime error or invalid argument would suit more.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

**Comment by PriyangaRamasamy** [comment]
rebased this PR on latest P11_Dev with the last patchset https://github.com/ibm-openbmc/openpower-vpd-parser/commit/8a8dc2945281c28412b68757e967ce876cb2c351

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by jinuthomas — COMMENTED** [review]
waiting for PR 291 , also restrict VHDR and VTOC for the time being , keep code commented , do not remove the code. retune as invalid arg for the time being .

File: include/ipz_parser.hpp [hunk 2]
```diff
@@ -75,6 +75,20 @@ class IpzVpdParser : public ParserInterface
...
 
+    /**
+     * @brief API to read keyword's value from hardware
+     *
+     * @param[in] i_paramsToReadData - Data required to perform read
+     *
+     * @throw
+     * sdbusplus::xyz::openbmc_project::Common::Device::Error::ReadFailure.
+     *
+     * @return On success return the value read from hardware. On failure
```
Comment (jinuthomas) [inline include/ipz_parser.hpp:86]: nit: same as below

Comment (PriyangaRamasamy) [reply include/ipz_parser.hpp]: done

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**